src/java/org/apache/cassandra/cql3/functions/OperationFcts.java

```diff
@@ -105,6 +106,22 @@ protected ByteBuffer executeOnNumerics(NumberType<?> resultType,
             {
                 return resultType.mod(leftType, left, rightType, right);
             }
+        },
+        CONCATENATION('+', "_concat")
```

Comment:
There is no need to add a new enum. You should simply implemente the  `excuteOnStrings` method in `ADD`. The internal name is hidden and does not play any important role.